### PR TITLE
chore: experiment actor refactor

### PR DESCRIPTION
**Pull request by erikwilson**
## Description

Removes actors from experiments.

## Test Plan

Automated testing.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-9661

### CODE REVIEW

**Review by carolinaecalderon — APPROVED** [review]
LGTM

File: master/internal/api_experiment.go [hunk 1]
```diff
@@ -1236,22 +1242,25 @@ func (a *apiServer) PatchExperiment(
...
+			e, ok := experiment.ExperimentRegistry.Load(int(exp.Id))
+			if !ok {
+				return nil, api.NotFoundErrs("experiment", fmt.Sprint(exp.Id), true)
+			}
+
 			if newResources.MaxSlots != nil {
-				a.m.system.TellAt(actor.Addr("experiments", int(exp.Id)),
-					sproto.SetGroupMaxSlots{MaxSlots: ptrs.Ptr(int(*newResources.MaxSlots))})
+				msg := sproto.SetGroupMaxSlots{MaxSlots: ptrs.Ptr(int(*newResources.MaxSlots))}
+				go e.SetGroupMaxSlots(msg)
```
Comment (stoksc) [inline master/internal/api_experiment.go:1252]: i don't think there is any good reason for this `go ..`, let's just make it sync.

File: master/internal/experiment/bulk_action.go [hunk 2]
```diff
@@ -277,11 +248,10 @@ func CancelExperiments(ctx context.Context, system *actor.System,
 		}
 	}
 
-	var refs []*actor.Ref
+	var refs []Experiment
```
Comment (stoksc) [inline master/internal/experiment/bulk_action.go:251]: nit, here and others, `var refs []Experiment` -> `refs := make(map[int]Experiment)` and 264 becomes `for id, ref := range refs` and we can save the `ref.GetID()`?

File: master/internal/experiment.go [hunk 3]
```diff
@@ -252,412 +209,463 @@ func newUnmanagedExperiment(
...
 	// Will only have the model, nothing required for the experiment actor.
-	return &experiment{
+	return &internalExperiment{
 		Experiment: expModel,
 	}, nil, nil
 }
 
-func (e *experiment) Receive(ctx *actor.Context) error {
+// StartAsync first registers the experiment and then starts asynchronously.
+func (e *internalExperiment) StartAsync() (*errgroup.Group, error) {
```
Comment (stoksc) [inline master/internal/experiment.go:218]: why do we think we can't safely move `StartAsync` into `new..` and always do it sync? or do we think we could do that as an immediate followup, at least?

**Review by stoksc — APPROVED** [review]
lgtm, though may be nice to chat about the future of that one tell that became a `go ...` call and `StartAsync`

File: master/internal/experiment.go [hunk 4]
```diff
@@ -252,412 +209,447 @@ func newUnmanagedExperiment(
...
-			return err
+	if err := e.register(); err != nil {
+		e.syslog.WithError(err).Error("failed to register experiment")
+		return err
+	}
+	defer func() {
+		if err != nil {
+			e.unregister()
 		}
+	}()
```
Comment (stoksc) [inline master/internal/experiment.go:230]: is a partial failure actually possible?

**Pull request merged**